builtin/providers/aws/resource_aws_nat_gateway.go

```diff
@@ -0,0 +1,169 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsNatGateway() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsNatGatewayCreate,
+		Read:   resourceAwsNatGatewayRead,
+		Delete: resourceAwsNatGatewayDelete,
+
+		Schema: map[string]*schema.Schema{
+			"allocation_id": &schema.Schema{
```

Comment:
Digging through the Go SDK, I found [this line](https://github.com/aws/aws-sdk-go/commit/1ddba7e17e1c7e6b30102a6d12a23e30133ccd08#diff-78f889de55d8e45bf3f86cdbee9f5f01R4576), which defines the `AllocationId` argument.

I did some searching and found the AWS CLI docs for allocating an EIP and the naming convention is already defined there in the [response output](https://docs.aws.amazon.com/cli/latest/reference/ec2/allocate-address.html#output).

I understand why the naming is what it is now, but I still think many folks will find it non-intuitive. What are your thought on changing it to `eip_allocation_id` to give it some more context?